src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -177,34 +183,72 @@ protected void LookupMembersInternal(LookupResult result, NamespaceOrTypeSymbol
         }
 
 #nullable enable
-        private void LookupExtensionMembersInSingleBinder(LookupResult result, TypeSymbol receiverType,
-            string name, int arity, ConsList<TypeSymbol>? basesBeingResolved, LookupOptions options,
-            Binder originalBinder, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        private void LookupAllExtensionMembersInSingleBinder(LookupResult result, TypeSymbol receiverType,
+            string? name, int arity, LookupOptions options, Binder originalBinder,
+            ref CompoundUseSiteInfo<AssemblySymbol> classicExtensionUseSiteInfo)
         {
-            Debug.Assert(name is not null);
+            var singleLookupResults = ArrayBuilder<SingleLookupResult>.GetInstance();
+            EnumerateAllExtensionMembersInSingleBinder(singleLookupResults, receiverType, name, arity, options, originalBinder, ref classicExtensionUseSiteInfo);
+            foreach (var singleLookupResult in singleLookupResults)
+            {
+                result.MergeEqual(singleLookupResult);
+            }
 
-            var extensions = ArrayBuilder<NamedTypeSymbol>.GetInstance();
+            singleLookupResults.Free();
+        }
 
-            Debug.Assert(!receiverType.IsDynamic());
-            if (!receiverType.IsErrorType())
+        internal void EnumerateAllExtensionMembersInSingleBinder(ArrayBuilder<SingleLookupResult> result,
+            TypeSymbol receiverType, string? name, int arity, LookupOptions options, Binder originalBinder,
```

Comment:
>TypeSymbol receiverType [](http://example.com/codeflow?start=12&length=23)

I think this parameter should be removed from this helper and and from all callers that simply propagate its value. One shouldn't be required to provide a receiver type in order to enumerate extensions.

Reply:
I like that! Thanks